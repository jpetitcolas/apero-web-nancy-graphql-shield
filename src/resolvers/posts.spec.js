const { gql } = require('apollo-server-core');
const { print } = require('graphql');
const superagent = require('superagent');

describe('Post Resolvers', () => {
    describe('getPosts', () => {
        const query = print(gql`
            query getPosts {
                posts {
                    id
                    title
                    author {
                        firstName
                        lastName
                        email
                    }
                }
            }
        `);

        it('should return a list of posts', async () => {
            const response = await superagent.post('http://localhost:5000/graphql')
                .send({ query, variables: {} });

            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
    });
});
