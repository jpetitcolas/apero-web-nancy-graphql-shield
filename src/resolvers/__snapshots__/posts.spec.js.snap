// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Resolvers getPosts should return a list of posts 1`] = `
Object {
  "data": Object {
    "posts": Array [
      Object {
        "author": Object {
          "email": "john.doe@example.com",
          "firstName": "John",
          "lastName": "Doe",
        },
        "id": "1",
        "title": "Tint is hiring!",
      },
      Object {
        "author": Object {
          "email": "john.doe@example.com",
          "firstName": "John",
          "lastName": "Doe",
        },
        "id": "2",
        "title": "Insurance: a $480B Addressable Market",
      },
      Object {
        "author": Object {
          "email": "alice.kellerman@example.com",
          "firstName": "Alice",
          "lastName": "Kellerman",
        },
        "id": "3",
        "title": "Take Part Into the Future of Insurance",
      },
    ],
  },
}
`;
